common/djangoapps/student/views.py

```diff
@@ -2486,18 +2486,22 @@ def validate_password(user, password):
         password: the user's proposed new password.
 
     Returns:
-        is_valid_password: a boolean indicating if the new password
-            passes the validation.
+        is_security_violated: a boolean indicating if the new password
+            passes the security policy validation.
         err_msg: an error message if there's a violation of one of the password
             checks. Otherwise, `None`.
     """
     err_msg = None
 
-    if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
-        try:
+    try:
+        if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
             validate_password_strength(password)
-        except ValidationError as err:
-            err_msg = _('Password: ') + '; '.join(err.messages)
+        else:
+            validate_password_length(password)
+
+    except ValidationError as err:
```

Comment:
@tasawernawaz I see, you need to know, if it's a strength based invalidation of the password, you need to provide the value in the `valid_link` variable below.
This means you should distinguish the validationError by it's `code` property.
This does explain why you need to return a `is_security_violated` flag. Can we rename that flag to something like `too_weak`?